fendermint/vm/interpreter/src/fvm/exec.rs

```diff
@@ -57,6 +57,8 @@ where
         // Block height (FVM epoch) as sequence is intentional
         let height = state.block_height();
 
+        self.gas.reset_block_gas_quota(&mut state)?;
```

Comment:
Why do we need a reset? We should be loading the current gas market state from the actor before we begin execution?

Reply:
Yep, ideally it's like this. I tried doing this as well, but the current `ABCI` trait is blocking me from doing this because it's `begin(&self, ...)`. If we reset everytime, then we have to do:
```rust
self.gas = GasMarket::try_from(&mut state)?;
```
which requires `begin(&mut self)`.

I tried not to update so many changes and keep the interface consistent.